packages/core/src/lib/connection_manager.ts

```diff
@@ -102,9 +103,10 @@ export class ConnectionManager {
 
         this.dialAttemptsForPeer.delete(peerId.toString());
         return;
-      } catch (error) {
-        log(`
-          Error dialing peer ${peerId.toString()}`);
+      } catch (error: any) {
```

Comment:
Can you cast the type to `Error` (I think that's the right one) instead?
ie,
```
const e = error as unknown as Error;
```

So that the right type can also be used in the `Map`?

Reply:
Sounds good to upgrade yes. re es2021 vs es2022 let's look at browser support and make a reasonable decision. Cc @weboko who might be more acquainted with those.